core/cloudflow-akka-tests/src/test/scala/cloudflow/akkastream/scaladsl/AkkaStreamletConsumerGroupSpec.scala

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.akkastream.util.scaladsl
+
+import scala.concurrent._
+import scala.concurrent.duration._
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.scaladsl._
+
+import com.typesafe.config._
+import cloudflow.akkastream._
+import cloudflow.akkastream.scaladsl._
+import cloudflow.akkastream.testdata._
+import cloudflow.streamlets._
+import cloudflow.streamlets.avro._
+
+import net.manub.embeddedkafka._
+import org.scalatest.time._
+
+object AkkaStreamletConsumerGroupSpec {
+  val kafkaPort = 1234
+  val zkPort    = 5678
+  val config    = ConfigFactory.parseString(s"""
+      akka {
+        stdout-loglevel = "OFF"
+        loglevel = "OFF"
+      }
+      cloudflow.kafka.bootstrap-servers = "localhost:$kafkaPort"
+      """)
+}
+
+import AkkaStreamletConsumerGroupSpec._
+
+class AkkaStreamletConsumerGroupSpec extends EmbeddedKafkaSpec(kafkaPort, zkPort, ActorSystem("test", config)) {
+
+  implicit override val patienceConfig = PatienceConfig(timeout = Span(10, Seconds), interval = Span(20, Millis))
+
+  override def createKafkaConfig: EmbeddedKafkaConfig =
+    EmbeddedKafkaConfig(kafkaPort,
+                        zooKeeperPort,
+                        Map(
+                          "broker.id"                        -> "1",
+                          "num.partitions"                   -> "53",
+                          "offsets.topic.replication.factor" -> "1",
+                          "offsets.topic.num.partitions"     -> "3"
+                        ))
+
+  "Akka streamlet instances" should {
+    "consume from an outlet as a group for the same streamlet" in {
+      // Generate some test data
+      val dataSize     = 5000
+      val data         = List.range(0, dataSize).map(i => Data(i, s"data"))
+      val outlet       = mkUniqueGenOutlet()
+      val genExecution = Generator.run(data, outlet)
+      // gen auto-completes, the source is finite.
+      genExecution.completed.futureValue
```

Comment:
We don't have to capture this future and/or map over it?

Reply:
Or is this some non-intuitive Scalatest way of asserting that the future completes successfully?

Reply:
OK, I read the docs now. IMO it would read better if we write it like this:

```suggestion
      val _ = genExecution.completed.futureValue // assert that the future completed
```

Simply because otherwise the syntax is so obtuse that it is hard to understand what is being done here.